src/metabase/driver/sql_jdbc/sync/describe_database.clj

```diff
@@ -95,7 +95,8 @@
       true
       (catch Throwable e
         (log/trace e "Assuming no SELECT privileges: caught exception")
-        (.rollback conn)
+        (when-not (.getAutoCommit conn)
+          (.rollback conn))
```

Comment:
I would think jdbc had a `inTransaction` method but it doesn't seem to. I believe we can't get here during  a manually opened transaction since `with-transaction` would catch and rollback on its own, but it might be worth checking.

Reply:
What do you mean by "we can't get here during a manually opened transaction"? I'm unsure about both "here" and where that "manually opened transaction" could be.

If we are in a transaction (manually opened or otherwise), we want to roll back, because the transaction cannot be reused for further checks otherwise. (That's why `.rollback` was added in the first place.) If we have auto commit enabled, there is no transaction to roll back and an exception is thrown. (This is what's being fixed here.) If a transaction is started (e.g., using `with-db-transaction`), then inside that transaction `.getAutoCommit` returns `false`, so we roll back.

What should be checked?

Reply:
> If a transaction is started (e.g., using with-db-transaction), then inside that transaction .getAutoCommit returns false, so we roll back.
I did not realize this happened, and thought the code was trying to prevent rolling back more than once.

```clojure
(toucan2.core/with-connection 
  [conn]
  (.setAutoCommit conn true)
  [(toucan2.core/with-transaction [tx conn]
     [(.getAutoCommit tx)
      (.getAutoCommit conn)])
   (.getAutoCommit conn)])
;; => [[false false] true]
```